### PR TITLE
chore: update flutter version to 3.24 and update dependencies

### PR DESCRIPTION
**Pull request by Feichtmeier**
Update to 3.24 stable

I added TODOs for the deprecated Navigator API, since we werent sure yet what to happen with YMDP.
I hope this is okay   @Jupi007 @spydon?

### CODE REVIEW

**Review by Jupi007 — APPROVED** [review]
LGTM :+1: 
Thanks @Feichtmeier !

**Comment by Jupi007** [comment]
> I added TODOs for the deprecated Navigator API, since we werent sure yet what to happen with YMDP.

That's a good idea, no need to spend time on something we could drop in a close future :)

**Comment by Feichtmeier** [comment]
Cool! I think the goldens will be fixed tonight
I checked progress indicator and it looks like always

https://github.com/user-attachments/assets/506395bd-744b-4e68-8503-5c5c944c9bca

File: example/lib/pages/full_color_icons_page.dart [hunk 1]
```diff
@@ -182,6 +182,7 @@ const iconDirsToIconNames = <String, Set<String>>{
     'accessories-dictionary',
     'accessories-text-editor',
     'address-book-app',
+    'app-center',
```
Comment (spydon) [inline example/lib/pages/full_color_icons_page.dart:185]: So we have a separate one called app-center? Since there is the snap-store icon too?

**Review by spydon — APPROVED** [review]
Lgtm, just one minor comment

File: example/lib/pages/full_color_icons_page.dart (hunk 1 above)
Comment (Feichtmeier) [reply example/lib/pages/full_color_icons_page.dart:185]: I just listed all app icons from yaru. Ideal would be some http call trying to list all those icons on its own (github dart api?) but I didnt manage this so it is a static list of all icons from yaru.

**Pull request merged**